docs/src/components/CombinationNew.js

```diff
@@ -0,0 +1,66 @@
+// @flow strict
+import { Flex } from 'gestalt';
+import React, { type Node } from 'react';
+import MainSectionCard from './MainSectionCard.js';
+
+type Props = {
+  children: (Object, number) => Node, // flowlint-line unclear-type:off
+  shaded?: boolean,
+  ...
+};
+
+const flatMap = (arr, fn) => arr.map(fn).reduce((a, b) => a.concat(b));
+const combinations = (variationsByField) => {
+  const fieldNames = Object.keys(variationsByField);
+
+  if (!fieldNames.length) return [{}];
+
+  const combine = ([fieldName, ...restFieldNames], acc) => {
+    const variationsForField = variationsByField[fieldName];
+
+    if (!Array.isArray(variationsForField) || !variationsForField.length) {
+      throw new Error(`Please provide a non-empty array of possible values for prop ${fieldName}`);
+    }
+
+    const vs = variationsForField.map((fieldValue) => ({
+      ...acc,
+      [fieldName]: fieldValue,
+    }));
+
+    if (!restFieldNames.length) {
+      return vs;
+    }
+    return flatMap(vs, (newAcc) => combine(restFieldNames, newAcc));
+  };
+
+  return combine(fieldNames, {});
+};
+
+const toReactAttribute = (key, value) => {
+  switch (typeof value) {
+    case 'boolean':
+      return (value && key).toString();
+    case 'string':
+      return `${key}=${JSON.stringify(value)}`;
+    default:
+      return `${key}={${JSON.stringify(value)}}`;
+  }
+};
+
+export default function CombinationNew({ children, shaded, ...props }: Props): Node {
```

Comment:
I don't have a great name for this component (eventually it will replace all uses of Combination)

Reply:
this works for now! or something like CombinationV2 — but it really doesn't matter since we'll delete Combination and rename this component in the near future